ui/app/common/concept-set/directives/formControls.js

```diff
@@ -79,6 +83,52 @@ angular.module('bahmni.common.conceptSet')
                     }
                 });
             };
+            var link = function ($scope, elem, attrs) {
+                $scope.$on('$stateChangeStart', function (event, next, current) {
+                    var navigating = next.url.split("/")[1];
+                    var allConsultationBoards = clinicalAppConfigService.getAllConsultationBoards();
+                    var outOfConsultationBoard = true;
+                    allConsultationBoards.map(function (board) {
+                        var consultationLink = board.url.split("/")[0];
+                        if (navigating.includes(consultationLink)) {
+                            outOfConsultationBoard = false;
+                        }
+                    });
+                    if (next.url.includes("/dashboard") && $state.params.patientUuid === current.patientUuid) {
+                        outOfConsultationBoard = false;
+                    }
+
+                    var checkAlreadyPresent = [];
+
+                    if ($scope.form.component && $scope.form.component.getValue().observations.length > 0) {
```

Comment:
A lot of code with multiple if-else blocks. Can we refactor this into smaller functions() / javascript objects.. and have some unit tests added?

Reply:
Sure